src/components/UserInfo/UserInfo.tsx

```diff
@@ -1,4 +1,20 @@
 // Add the required types and props
-export const UserInfo = () => (
-  <>UserInfo markup</>
+import { User } from '../../types/User';
+
+type Props = {
+  user: User | null;
+};
+
+export const UserInfo: React.FC<Props> = ({ user }) => (
+  <>
+    {user && (
+      <a
+        className="UserInfo"
+        href={`mailto:${user.email}`}
+        data-id={user.id}
+      >
+        {user.name}
+      </a>
+    )}
+  </>
```

Comment:
move ckecking to the up
```suggestion
  <a
    className="UserInfo"
    href={`mailto:${user.email}`}
    data-id={user.id}
  >
    {user.name}
  </a>
```